packages/model-viewer/src/features/scene-graph.ts

```diff
@@ -210,8 +204,22 @@ export const SceneGraphMixin = <T extends Constructor<ModelViewerElementBase>>(
           shadow.visible = false;
         }
 
-        const exporter = new GLTFExporter();
-        exporter.parse(scene.modelContainer, (gltf) => {
+        const currentGLTF = this[$currentGLTF];
+
+        if (currentGLTF != null && 'functions' in currentGLTF.userData &&
+            'ensureLoadVariants' in currentGLTF.userData.functions) {
+          // Ensure all variant materials are loaded because some of them may
+          // not be loaded yet.
+          await currentGLTF.userData.functions.ensureLoadVariants(scene);
```

Comment:
The loader plugin loads variant materials on demand. The exporter plugin doesn't export unloaded variant materials so ensuring all variant materials are loaded here.